### PR TITLE
Fix: visual glitch with autocomplete component

### LINKED ISSUE #20139: Fix visual glitch with autocomplete component
Reported by @rt4914 on [this PR](https://github.com/forem/forem/pull/20123). There's a visual glitch when interacting with autocomplete components, possibly when they layer "on top of" other UI elements. Initially seen in `UsernameInput` but likely effects tags and at-mentions as well, as it's likely an issue in the underlying component. It's hard to put into words, possibly best captured in this video:
https://www.loom.com/share/b410b5fc7829431baf1c20036eb1c1c1?sid=f6503ec0-b374-4fba-b5d3-bea3b5f35ed9

### PR DESCRIPTION
**Pull request by morfenza**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Fixes the visual glitch by forcing a display block inside span tag in the component. 

I tried to isolate the dropdown to change its properties, I thought that the best way was to use the style attribute inside its own tag. Fisrtly I used a z-index value that was higher than anything else, but I didn't think it solved the flicking showed in the video, although it seemed better it still looked weird in how it appeared and covered half the button, so I used a display block cover the button the same way the dropdown selection does, it also removed the flicking problem.

Although it originally seemed to solve the visual glitch, for a few frames, it still appeared behind the button, so I changed the approach to change the dropdown position to relative and decrease the max-height so it does not disrupts the page layout when there are a lot of moderators.

## Related Tickets & Documents

- Closes #20139

### System Specs:
* OS: Manjaro Linux x86_64 
* Browser: Firefox 122.0 (64-bits)
* I've used dip to run the application locally

Peek 2024-02-20 16-37

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [X] No, they are not needed

### CODE REVIEW

**Comment by joaoGabriel55** [comment]
Watching the GIF, I could see that at a specific moment, the dropdown menu goes to the back of the button.

**Comment by morfenza** [comment]
> Watching the GIF, I could see that at a specific moment, the dropdown menu goes to the back of the button.

That's true.. I'll investigate it

**Comment by morfenza** [comment]
New solution found with the help of @joaoGabriel55! Thx!

File: app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx [hunk 1]
```diff
@@ -650,7 +650,7 @@ export const MultiSelectAutocomplete = ({
         </div>
         {showMaxSelectionsReached ? (
           <div className="c-autocomplete--multi__popover">
-            <span className="p-3">Only {maxSelections} selections allowed</span>
+            <span className="p-3">Only {maxSelections} selection allowed</span>
```
[the remark this answers is not in the document]
Comment (morfenza) [reply app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx:653]: Nice! I'll include it!

**Pull request merged**